src/iceberg/partition_spec.cc

```diff
@@ -177,6 +177,41 @@ Status PartitionSpec::Validate(const Schema& schema, bool allow_missing_fields)
   return {};
 }
 
+Status PartitionSpec::ValidatePartitionName(const Schema& schema) const {
```

Comment:
Do you think we need to call this function in the `PartitionSpec::Validate` as well?